src/main.cpp

```diff
@@ -2577,7 +2577,7 @@ bool FindBlockPos(CValidationState &state, CDiskBlockPos &pos, unsigned int nAdd
         pos.nPos = vinfoBlockFile[nFile].nSize;
     }
 
-    if (nFile != nLastBlockFile) {
```

Comment:
This is obviously the smaller change, but,  couldn't `nLastBlockFile` just be `unsigned`?